dataRetriever/resolvers/validatorInfoResolver_test.go

```diff
@@ -361,6 +440,198 @@ func TestValidatorInfoResolver_ProcessReceivedMessage(t *testing.T) {
 		assert.Nil(t, err)
 		assert.True(t, wasCalled)
 	})
+
+	// resolveMultipleHashesRequest
+	t.Run("unmarshal fails", func(t *testing.T) {
+		t.Parallel()
+
+		numCalls := 0
+		expectedErr := errors.New("expected err")
+		args := createMockArgValidatorInfoResolver()
+		args.Marshaller = &testscommon.MarshalizerStub{
+			UnmarshalCalled: func(obj interface{}, buff []byte) error {
+				marshallerMock := testscommon.MarshalizerMock{}
+				if numCalls < 1 {
```

Comment:
so the first unmarshal is for request data which should work, and you want to simulate the unmarshalling failing for the batch, right? might add a comment

Reply:
or you can use a type switch like this:
```golang
UnmarshalCalled: func(obj interface{}, buff []byte) error {
				switch obj.(type) {
				case *dataRetriever.RequestData:
					return testscommon.MarshalizerMock{}.Unmarshal(obj, buff)
				case *batch.Batch:
					return expectedErr
				}
				return nil
			},
```

Reply:
this is a way better looking solution. thanks, fixed